checker/parsing/utils.py

```diff
@@ -120,73 +120,63 @@ class _EvaluateFalseTransformer(sympy_parser.EvaluateFalseTransformer):
        future proof!
     """
 
-    _evaluate_false_keyword = ast.keyword(arg='evaluate', value=ast.Name(id='False', ctx=ast.Load()))
+    _evaluate_false_keyword = ast.keyword(arg='evaluate', value=ast.NameConstant(value=False, ctx=ast.Load()))
     _one = ast.Call(func=ast.Name(id='Integer', ctx=ast.Load()), args=[ast.Num(n=1)], keywords=[])
     _minus_one = ast.UnaryOp(op=ast.USub(), operand=_one)
     _bool_ops = {
         ast.And: "And",
         ast.Or: "Or"
     }
     _bool_values = {
-        True: ast.Name(id='True', ctx=ast.Load()),
-        False: ast.Name(id='False', ctx=ast.Load())
+        True: ast.Call(func=ast.Name(id="true", ctx=ast.Load()), args=[], keywords=[]),
+        False: ast.Call(func=ast.Name(id="false", ctx=ast.Load()), args=[], keywords=[])
     }
+    _ignore_functions = ["Integer", "Float", "Symbol", "factorial"]
 
     def visit_Call(self, node):
         """Ensure all function calls are 'evaluate=False'."""
-        # Since we have overridden the visit method, we are now responsible for
-        # ensuring all child nodes are visited too. This is done most simply by
-        # calling generic_visit(...) on ourself:
-        self.generic_visit(node)
         # FIXME: Some functions cannot accept "evaluate=False" as an argument
         # without their __new__() method raising a TypeError. There is probably
         # some underlying reason which we could take into account of.
-        # For now, blacklist those known to be problematic:
-        _ignore_functions = ["Integer", "Float", "Symbol", "factorial"]
-        if node.func.id in _ignore_functions:
-            # print("\tIgnoring function: {}".format(node.func.id))
-            pass
-        else:
-            # print("\tModifying function: {}".format(node.func.id))
-            node.keywords.append(self._evaluate_false_keyword)
-        # We must return the node, modified or not:
-        return node
+        # For now, blacklist those known to be problematic (in self._ignore_functions)
+        new_node = self.generic_visit(node)
+        if isinstance(node.func, ast.Name) and node.func.id not in self._ignore_functions:
+            new_node.keywords.append(ast.keyword(arg='evaluate', value=ast.NameConstant(value=False, ctx=ast.Load())))
```

Comment:
```suggestion
            new_node.keywords.append(self._evaluate_false_keyword)
```
The attribute exists to remove this duplication, and this is what it used to be.